test/e2e/framework/framework.go

```diff
@@ -0,0 +1,134 @@
+// Copyright 2018 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package framework is a package helping setting up end-to-end testing accross a
+// Kubernetes cluster.
+package framework
+
+import (
+	"fmt"
+	"net"
+	"time"
+
+	"agones.dev/agones/pkg/apis/stable/v1alpha1"
+	"agones.dev/agones/pkg/client/clientset/versioned"
+	"github.com/pkg/errors"
+	"k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/kubernetes"
+	// required to use gcloud login see: https://github.com/kubernetes/client-go/issues/242
+	_ "k8s.io/client-go/plugin/pkg/client/auth/gcp"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+// Framework is a testing framework
+type Framework struct {
+	KubeClient      kubernetes.Interface
+	AgonesClient    versioned.Interface
+	GameServerImage string
```

Comment:
So I'm trying to work out what this is used for? It doesn't look like it gets used?

Reply:
you're right, it should be used here https://github.com/Kuqd/agones/blob/e4f921d744fc8117c97dc3efd71a0b3bdb8be247/test/e2e/gameserver_test.go#L45, do you want me to create a getter in the framework ? or should I move the creation to the framework ?